src/Command/SubProfileCommand.php

```diff
@@ -0,0 +1,363 @@
+use Drupal\Console\Command\Shared\ConfirmationTrait;
+use Drupal\Console\Command\Shared\FormTrait;
+use Drupal\Console\Command\Shared\ModuleTrait;
+use Drupal\Console\Core\Command\Shared\CommandTrait;
+use Drupal\Console\Core\Style\DrupalStyle;
+use Drupal\Console\Core\Utils\ConfigurationManager;
+use Drupal\Console\Core\Utils\FileQueue;
+use Drupal\Console\Core\Utils\StringConverter;
+use Drupal\Console\Core\Utils\TwigRenderer;
+use Drupal\Console\Core\Utils\TranslatorManager;
+use Drupal\Console\Extension\Manager;
+use Drupal\Console\Utils\Site;
+use Drupal\Console\Utils\Validator;
+use Drupal\Core\Extension\ExtensionDiscovery;
+use Drupal\Core\Extension\InfoParser;
+use Drupal\lightning\Generator\SubProfileGenerator;
+use Drupal\lightning_core\Element;
+use GuzzleHttp\Client;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+
+class SubProfileCommand extends ProfileCommand {
+
+  use CommandTrait;
+  use ModuleTrait;
+  use FormTrait;
+  use ConfirmationTrait;
+
+  /**
+   * @var string \Drupal::root().
+   */
+  protected $appRoot;
+
+  /**
+   * @var array
+   */
+  protected $topLevelComponents;
+
+  /**
+   * @var array
+   */
+  protected $excludedDependencies;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function configure() {
+    $this
+      ->setName('lightning:subprofile')
+      ->setDescription($this->trans('Generate a subprofile of Lightning'))
+      ->setHelp($this->trans('The <info>lightning:subprofile</info> command helps you generate a new subprofile of Lightning'))
+      ->addOption(
+        'profile',
+        '',
+        InputOption::VALUE_REQUIRED,
+        $this->trans('The name of the subprofile')
+      )
+      ->addOption(
+        'machine-name',
+        '',
+        InputOption::VALUE_REQUIRED,
+        $this->trans('The machine name of the subprofile (lowercase and underscore only)')
+      )
+      ->addOption(
+        'description',
+        '',
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('Subprofile description')
+      )
+      ->addOption(
+        'core',
+        '',
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('commands.generate.profile.options.core'),
+        '8.x'
+      )
+      ->addOption(
+        'dependencies',
+        false,
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('commands.generate.profile.options.dependencies')
+      )
+      ->addOption(
+        'excluded_dependencies',
+        false,
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('Top-level components of Lightning to exclude separated by commas.')
+      )
+      ->addOption(
+        'excluded_subcomponents',
+        false,
+        InputOption::VALUE_OPTIONAL,
+        $this->trans('Lightning sub-components to exclude separated by commas.')
+      )
+      ->addOption(
+        'distribution',
```

Comment:
As we discussed the other day, we should probably remove this option entirely. Anyone seeking to create a distribution can do that manually.